test/lib/redis-store-zlib-spec.js

```diff
@@ -0,0 +1,372 @@
+var config = require('../config.json');
+var redisStore = require('../../index');
+var sinon = require('sinon');
+var assert = require('assert');
+var zlib = require('zlib');
+
+var redisGzipCache;
+var customRedisGzipCache;
+var testJson;
+
+describe('Gzip Tests', function () {
+
+  before(function () {
+    redisGzipCache = require('cache-manager').caching({
+      store: redisStore,
+      host: config.redis.host,
+      port: config.redis.port,
+      auth_pass: config.redis.auth_pass,
+      db: config.redis.db,
+      ttl: config.redis.ttl,
+      gzip: true
```

Comment:
Now thinking about it, I wanted to ask what you guys @PuKoren @jeff-kilbride would think about making this a little bit more abstract. To be able to easily plug in other compressors/decompressors and gzip being the first choice for now.

So instead of having gzip:true having a `compression: 'gzip'`(or default to it if compression: true). Based on the value of compression property user can pick which one they need (maybe in future there would be others) or do their own.  I know that this means a little big refactor, but it should open up options. (maybe compression is a bad name, but we can adapt and chose another)